File: src/main/kotlin/Java/Escuela/Profesor.java
package Java.Escuela;

class Profesor {
    private String nombre;

    public Profesor(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() {
        return nombre;
    }

    public void asignarCurso(Estudiante estudiante, Curso curso) {
        estudiante.agregarCurso(curso);
    }

    public void calificarEstudiante(Curso curso, Estudiante estudiante, double calificacion) {
        curso.asignarCalificacion(estudiante, calificacion);
    }

    @Override
    public String toString() {
        return nombre;
    }
}

File: src/main/kotlin/kotlinOPP/Escuela/Profesor.kt
package kotlinOPP.Escuela

class Profesor(val nombre: String) {
    fun asignarCurso(estudiante: Estudiante, curso: Curso) {
        estudiante.agregarCurso(curso)
    }

    fun calificarEstudiante(curso: Curso, estudiante: Estudiante, calificacion: Double) {
        curso.asignarCalificacion(estudiante, calificacion)
    }

    override fun toString(): String {
        return nombre
    }
}